website/src/blogs/2017-10-18-how-to-build-an-e2e.md

```diff
@@ -0,0 +1,80 @@
+---
+title: How to build an e2e?
+author: Amit Kumar Das
+author_info: Engineer the DAO
+tags: Cloud Storage, E2e Testing, Featured, Kubernetes, Storage
+date: 17-10-2017
+excerpt: e2e which expands into end to end speaks for itself. It can be treated as component testing, integration testing, or something that tries to test stuff outside the scope of unit tests.
+---
+
+### What is e2e?
+
+Well *e2e* which expands into end to end speaks for itself. It can be treated as component testing, integration testing, or something that tries to test stuff outside the scope of unit tests. I got attracted to this side of development when I was surprised looking at one of the smallest possible directory names ever in a code-based project. As I speak, we get to see a number of Go based projects having an *e2e* folder. I believe, it has been popularized by the likes of etcd, Kubernetes, and perhaps a few other open source communities.
+
+### How to build one?
+
+Before even getting into the How’s part, we must rather be comfortable with following questions:
+
+— *Do we need to ?*  
+ — *Does our project need one ?*
+
+More often than not, a code based project will love to have one. Perhaps not with this name but the need will definitely exist. Some of the guiding principles highlighted in this article will also help us in getting the answers to these questions. So keep reading.
+
+### Assuming our project needs one, how to build it then?
+
+Are there any guidelines / rules that we can follow ? Here they are:
+
+— ***`Refer and re-use wherever possible than build from scratch`*** is the golden rule.
+
+— ***`Start Small`***. In this context it refers to '*not to gulp the entire code base and be very selective even to the point of refusing 90% of the code that is available for free*'. This is the learning I have had all these years as a programmer. This is essential if one wants to reach the milestones in time which will otherwise become a mirage. I have burnt my fingers multiple times with regards to this principle. The latest one was when I tried to reuse the entire Kubernetes e2e for [Maya’s](https://github.com/openebs/maya/) e2e. Needless to say I failed miserably. Some of these reasons are explained in this article.
```

Comment:
```suggestion
— ***` Start Small`***. In this context, it refers to '*not to gulp the entire code base and be very selective even to the point of refusing 90% of the code that is available for free*'. This is the learning I have had all these years as a programmer. This is essential if one wants to reach the milestones in time which will otherwise become a mirage. I have burnt my fingers multiple times with regards to this principle. The latest one was when I tried to reuse the entire Kubernetes e2e for [Maya’s](https://github.com/openebs/maya/) e2e. Needless to say I failed miserably. Some of these reasons are explained in this article.
```